### ISSUE TITLE
Implement more realistic service to demonstrate better Behat usage

### ISSUE THREAD

**Issue by asgrim** [asgrim/quality-tutorial#6]
Feedback from Steve was that he left feeling that Behat didn't hold much over phpunit: so whilst I got the point across that you should be testing at the "domain" level, because the code we were testing was too contrived, there wasn't any "complexity" to make the domain testing "shine" as it were. Could improve this by having a more realistic service implementation for future.

**Comment by Ocramius** [comment]
Note that what should emerge from all the Behat discussion is that there is a readable specification/documentation of the software.

The testing bit is more like a side-effect of that :-)

**Comment by asgrim** [comment]
Yeah I think I got that bit across, was mainly the implementation of the tests was a bit sparse (maybe my domain implementation was "perfect" hahahahahahah)
